controllers/cinder_controller.go

```diff
@@ -911,6 +933,23 @@ func (r *CinderReconciler) schedulerDeploymentCreateOrUpdate(ctx context.Context
 }
 
 func (r *CinderReconciler) backupDeploymentCreateOrUpdate(ctx context.Context, instance *cinderv1beta1.Cinder) (*cinderv1beta1.CinderBackup, controllerutil.OperationResult, error) {
+
+	bakTemplate := instance.Spec.CinderBackup
+	cinderBackupSpec := cinderv1beta1.CinderBackupSpec{
+		ContainerImage:             bakTemplate.ContainerImage,
+		Replicas:                   bakTemplate.Replicas,
+		Debug:                      bakTemplate.Debug,
+		CustomServiceConfig:        bakTemplate.CustomServiceConfig,
+		CustomServiceConfigSecrets: bakTemplate.CustomServiceConfigSecrets,
+		DefaultConfigOverwrite:     bakTemplate.DefaultConfigOverwrite,
+		NetworkAttachments:         bakTemplate.NetworkAttachments,
+		ServiceUser:                instance.Spec.ServiceUser,
+		DatabaseHostname:           instance.Status.DatabaseHostname,
+		DatabaseUser:               instance.Spec.DatabaseUser,
+		Secret:                     instance.Spec.Secret,
+		ExtraMounts:                instance.Spec.ExtraMounts,
+		TransportURLSecret:         instance.Status.TransportURLSecret,
+	}
```

Comment:
kuttl deployment failed because it's not propagating c-bak's PasswordSelectors

Reply:
Thanks, Good catch, I have missed that in bak, and this probably highlight the need of avoiding code duplication whenever is possible (it would avoid this kind of issues).